packages/tooltip/src/tooltipUtils.tsx

```diff
@@ -0,0 +1,105 @@
+import { css } from '@leafygreen-ui/emotion';
+
+export function trianglePosition(
+  alignment: string,
+  justification: string,
+  triggerRect: DOMRect | ClientRect | null,
+) {
+  if (!alignment || !justification || !triggerRect) {
+    return '';
+  }
+
+  const borderOffset = 'calc(100% - 1px)';
+
+  const containerSize = 15;
+  const notchSize = 8;
+  const containerOffsetX = (triggerRect.width - containerSize) / 2;
+  const containerOffsetY = (triggerRect.height - containerSize) / 2;
+  const notchOverlap = -notchSize / 2;
+  const containerStyleObj: any = {};
+  const notchStyleObj: any = {};
+
+  switch (alignment) {
+    case 'top':
+    case 'bottom':
+      notchStyleObj.left = 0;
+      notchStyleObj.right = 0;
+
+      if (alignment === 'top') {
+        containerStyleObj.top = `${borderOffset}`;
+        notchStyleObj.top = `${notchOverlap}px`;
+      } else {
+        containerStyleObj.bottom = `${borderOffset}`;
+        notchStyleObj.bottom = `${notchOverlap}px`;
+      }
+
+      switch (justification) {
+        case 'left':
+          containerStyleObj.left = `${containerOffsetX}px`;
+          break;
+
+        case 'center-horizontal':
+          containerStyleObj.left = '0px';
+          containerStyleObj.right = '0px';
+          containerStyleObj.margin = 'auto';
+          break;
+
+        case 'right':
+          containerStyleObj.right = `${containerOffsetX}px`;
+          break;
+      }
+
+      break;
+
+    case 'left':
+    case 'right':
+      notchStyleObj.top = 0;
+      notchStyleObj.bottom = 0;
+
+      if (alignment === 'left') {
+        notchStyleObj.left = `${notchOverlap}px`;
+
+        // right align is .8px off, factoring that in here to correct offset
```

Comment:
Hmm... There has to be a better way. That discrepancy is likely to change based on other variables, so this is brittle.

Reply:
Agreed, just not sure what that .8 is from yet and this was my best attempt for now.

Reply:
If you change a variable like notchSize, does this discrepancy change?

Reply:
nope...